server/models.py

```diff
@@ -524,12 +524,9 @@ def enroll_from_form(cid, form):
         }
         Enrollment.create(cid, [info], role)
 
-    @staticmethod
     @transaction
-    def unenroll(enrollment):
-        db.session.delete(enrollment)
-        db.session.commit()
-        cache.delete_memoized(User.is_enrolled)
+    def unenroll(self):
+        db.session.delete(self)
```

Comment:
I think it's better if keep the cache clearing cache here (and not in the route) - so that others can use this method without forgetting to clear the cache.